src/rules.rs

```diff
@@ -233,89 +303,32 @@ pub fn get_all_rules() -> Vec<Box<dyn LintRule>> {
   ]
 }
 
-pub fn get_recommended_rules() -> Vec<Box<dyn LintRule>> {
-  get_all_rules()
-    .into_iter()
-    .filter(|r| r.tags().contains(&"recommended"))
-    .collect()
-}
-
-/// Returns a list of rules after filtering.
-///
-/// Following rules are applied (in the described order):
-///
-/// - if `maybe_tags` is `None` then all defined rules are returned, otherwise
-///   only rules matching at least one tag will be returned; if provided list
-///   is empty then all rules will be excluded by default
-///
-/// - if `maybe_exclude` is `Some`, all rules with matching codes will
-///   be filtered out
-///
-/// - if `maybe_include` is `Some`, rules with matching codes will be added
-///   to the return list
-///
-/// Before returning the list will sorted alphabetically.
-pub fn get_filtered_rules(
-  maybe_tags: Option<Vec<String>>,
-  maybe_exclude: Option<Vec<String>>,
-  maybe_include: Option<Vec<String>>,
-) -> Vec<Box<dyn LintRule>> {
-  let tags_set =
-    maybe_tags.map(|tags| tags.into_iter().collect::<HashSet<_>>());
-
-  let mut rules = get_all_rules()
-    .into_iter()
-    .filter(|rule| {
-      let mut passes = if let Some(tags_set) = &tags_set {
-        rule
-          .tags()
-          .iter()
-          .any(|t| tags_set.contains(&t.to_string()))
-      } else {
-        true
-      };
-
-      if let Some(excludes) = &maybe_exclude {
-        if excludes.contains(&rule.code().to_owned()) {
-          passes &= false;
-        }
-      }
-
-      if let Some(includes) = &maybe_include {
-        if includes.contains(&rule.code().to_owned()) {
-          passes |= true;
-        }
-      }
-
-      passes
-    })
-    .collect::<Vec<_>>();
-
-  rules.sort_by_key(|r| r.code());
-
-  rules
-}
-
 #[cfg(test)]
 mod tests {
   use super::*;
 
   #[test]
   fn recommended_rules_sorted_alphabetically() {
-    let mut recommended_rules = get_recommended_rules();
-    recommended_rules.sort_by_key(|r| r.code());
-    for (sorted, unsorted) in
-      recommended_rules.into_iter().zip(get_recommended_rules())
+    let sorted_recommended_rules = {
+      let mut recommended_rules =
+        Arc::try_unwrap(get_recommended_rules()).unwrap();
```

Comment:
A bunch of `#[derive(Debug)]` for all the lint structs are required to call `unwrap` here.